### PR TITLE
guide on troubleshooting network issues

### PR DESCRIPTION
**Pull request by astra1993**

### CODE REVIEW

File: network-troubleshooting-guide.en.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Network Troubleshooting Guide
+
+C&C Online can offer some of the most engaging and rewarding experiences—whether you're teaming up with friends for co-op or competing in a 2v2 match online. However, smooth online play depends heavily on stable and reliable network connections. Inconsistent latency, dropped connections, and server issues can quickly turn a fun session into a frustrating one.
+
+This guide is designed to help players diagnose and resolve common network-related problems that may arise when trying to connect to multiplayer services across the Command & Conquer titles supported by the C&C Online service.
+
+These games were released a long time ago, therefore some problems tend to come up in modern networking environments. Problems can range from not being able to connect to the servers to failing to join lobbies or start a multiplayer game, and unfortunately there are multiple causes to each issue, making the troubleshooting process complicated.
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change:

therefore some problems tend to come up in modern networking environments.

To:

These games were released a long time ago; therefore, some problems tend to come up in modern networking environments.

And:

and unfortunately there are multiple causes to each issue

To: and, unfortunately, each issue can have multiple causes

Comment (astra1993) [reply network-troubleshooting-guide.en.md]: fixed! :pray:

File: network-troubleshooting-guide.en.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Checking whether you have an IPv4 assigned to you is actually very easy. Just visit one of the websites that show you your IP address (e.g. [this site](https://whatismyipaddress.com) or just google for another website), and check if you can see an IPv4 value.
+
+![checking for an external IPv4 address](assets/for/network-troubleshooting-guide.en/check-ipv4.jpg)
+
+All of the games supported by us establish peer-to-peer (P2P) connections for their multiplayer, and the central server is only used to allow the players to find each other. The only protocol the games use for P2P is UDP, and they do UDP Hole Punching for NAT Traversal. This means that most players will not need any form of port forwarding or DMZ, as the games correctly negotiates their inbound ports. Also note that our games DO NOT negotiate their port forwardings through UPnP, so it being enabled or disabled has no effect on the connectivity (unless you use an 3rd-party UPnP client and use a static port which is taught in this guide later on).
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change: 

use an 3rd-party UPnP client

To:

use a 3rd-party UPnP client

Comment (astra1993) [reply network-troubleshooting-guide.en.md]: fixed!

File: network-troubleshooting-guide.en.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+If the ISP's CG-NAT implementation is not NAT traversal (NAT-T) capable (meaning it doesn't support UDP hole punching), players behind it will often be unable to host or directly connect to other players. This can result in connection timeouts, failure to join lobbies or start games, or unstable gameplay sessions. In essence, the network is unable to negotiate a direct path between peers, breaking the core mechanism that P2P multiplayer depends on.
+
+You can check whether you are behind a CG-NAT by following this procedure:
+
+1) Log into your modem/router.
+2) Usually the very first page that comes up shows your connection information (otherwise look for a network status page). You can find the IPv4 assigned to you there.
+3) If the IP starts with `10`, `192`, `172`, or `100` (which differs from your external IP which you found earlier in the guide), then you are behind a CG-NAT.
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change:

your external IP which you found earlier in the guide

TO:

your external IP, which you found earlier in the guide,

Comment (astra1993) [reply network-troubleshooting-guide.en.md]: fixed!

File: network-troubleshooting-guide.en.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+1) Start the game.
+2) Open the Tacitus overlay, by pressing its hotkey `Ctrl+T`.
+3) Go to the `Network` tab in the overlay.
+4) Enable the `UseAltPeerChatPort` option.
+5) Press `Save` and use `Ctrl+T` to close the overlay.
+6) Try logging in again, to see if the issue is resolved.
+
+### Edge Case: Red Alert 3 Firewall Port Override
+
+There is a setting in Red Alert 3, which is known to cause problems for setting up multiplayer games (the error is usually a `Failed Connection 1-2`). This is caused by setting a static Port Number in the game's settings under the Network section. Unfortunately just removing the port does not resolve the issue, and its directive must be manually removed from the game's options file.
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change:

caused by setting a static Port Number

To:

caused by setting a static port number

Capitilasation of the P and N is not correct. (however to use it to highlight in the text, if you really want. go for it i guess :p)

Comment (astra1993) [reply network-troubleshooting-guide.en.md]: fixed!

File: network-troubleshooting-guide.en.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+The second option is the `WarpPort` tool, which can be downloaded from [here](https://warpport.kaneswrath.com). This tool provides multiple tests, but the only tests you need to do are the `Hole Punch Test` and the `Peer-to-Peer Test`. It is noteworthy that the `Peer-to-Peer Test`, requires both players to have the tool installed, in order to run the test. There is an instructional video available on the provided link, that teaches you how to run these two tests.
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change:

the Peer-to-Peer Test, requires both players

To:

the Peer-to-Peer Test requires both players

You dont need that comma.

File: network-troubleshooting-guide.en.md (hunk 4 above)
Comment (astra1993) [reply network-troubleshooting-guide.en.md]: it was for highlighting, but isn't necessary. switched it to lowercase

File: network-troubleshooting-guide.en.md (hunk 5 above)
Comment (astra1993) [reply network-troubleshooting-guide.en.md]: :+1:

File: network-troubleshooting-guide.en.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+If you have activated a VPN interface, you must select the VPN adapters local IP from this list.
+
+In this section, there is also an option called `Send Delay`. It was originally designed to help with some D-Link and Netgear modems/routers, but we have observed it helping with situations involving different connection issues as well.
+
+**NOTE:** There is a rare edge-case, where the presence of a virtual network adapter on the system, that is disconnected but not disabled prevents the game from doing NAT traversal correctly. In our tests, Port Forwarding solved this problem, but the better approach is to disable the adapter when it's not needed. Selecting the correct in-game IP address from the settings might also remedy this problem.
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change:

might also remedy this problem.

To: might also resolve this problem.

Not everyone knows the word remedy, prehaps sounding a bit more "simple is better" 
I am not entertaily certain here you decide on this :p

Comment (astra1993) [reply network-troubleshooting-guide.en.md]: fixed. the meaning is preserved either way, so the more frequent word is the simpler and better option

File: network-troubleshooting-guide.en.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+These tools simulate real-world connection attempts between peers and measure how your network responds to outbound and inbound UDP traffic. By doing so, they can detect whether your router, firewall, or ISP is allowing the types of connections required for smooth P2P gameplay.
+
+We strongly encourage players experiencing frequent connection failures or hosting issues to run these tools before going for the other more complicated methods.
+
+It is also very important to remember, that if two people are failing to connect to each other, the issue might be with just one user and not the other. Always test your connectivity with people who are known to have correct working connection (you can ask the tech support staff for a test). This will allow the players to pinpoint directly which user has connectivity issues.
+
+It is worth mentioning, that these tools can also be used to test whether a VPN provider supports the connectivity features required by the games. So if you want to give a VPN provider, which isn't tested by us, a try, you can use these tools to test them.
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md]: Change:

So if you want to give a VPN provider, which isn't tested by us, a try

To:

So if you want to try a VPN provider that we haven’t tested,

File: network-troubleshooting-guide.en.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+This guide is designed to help players diagnose and resolve common network-related problems that may arise when trying to connect to multiplayer services across the Command & Conquer titles supported by the C&C Online service.
+
+These games were released a long time ago, therefore some problems tend to come up in modern networking environments. Problems can range from not being able to connect to the servers to failing to join lobbies or start a multiplayer game, and unfortunately there are multiple causes to each issue, making the troubleshooting process complicated.
+
+Please read through all of the sections in order, and try all the troubleshooting methods before reaching out to the support team, as this will allow us to pinpoint edge cases and unknown issues and focus our effort more effectively.
+
+### Preliminary Considerations
+
+The games supported by our service are:
+
```
Comment (CWedvin) [inline network-troubleshooting-guide.en.md:14]: * Command & Conquer: Generals (CCG)
* Command & Conquer: Generals: Zero Hour (ZH)
* Command & Conquer 3: Tiberium Wars (TW)
* Command & Conquer 3: Kane's Wrath (KW)
* Command & Conquer: Red Alert 3 (RA3)

Adding it after the naming might help with clearing up what is what in the abrivations, 
It might be a tad to much but then again we are dealing with people who might be stupid as fuck :p

File: network-troubleshooting-guide.en.md (hunk 7 above)
Comment (astra1993) [reply network-troubleshooting-guide.en.md]: fixed

**Review by CWedvin — CHANGES_REQUESTED** [review]
Take a look at my comments and decide for your self. 

When you made changes or if you did not, then make another request

File: network-troubleshooting-guide.en.md (hunk 8 above)
Comment (astra1993) [reply network-troubleshooting-guide.en.md:14]: good point. fixed

**Comment by astra1993** [comment]
all of the changes are applied and committed

**Review by CWedvin — APPROVED** [review]
Should there something else, we can simply change it again xD
